MINOR Enable the lazy loading for DependentPages GridField

maxime-rainville opened the pull request:
Fix https://github.com/silverstripe/silverstripe-cms/issues/2276

Tweak the Dependent page GridField to use `GridFieldLazyLoader`.

robbieaverill commented on code/Model/SiteTree.php:
```diff
@@ -1914,6 +1916,7 @@ public function getCMSFields()
                         );
                     }
                 ));
+            $dependentTable->getConfig()->addComponent(new GridFieldLazyLoader());
```
Please use Injector to instantiate classes

maxime-rainville replied:
None of the Gridfield components are injectable. Not sure if there's an underlying reason for this or if it's just an omission.

If we think it would be better to make them injectable, then we should probably have a separate job for this.

robbieaverill replied:
Yep, that's right. I'm happy to have a separate issue for it though

maxime-rainville commented:
Addressed all points. I've also remove the Absolute Link column following some discussion with Ingo.

I've created a follow up issue we could consider at a later date. https://github.com/silverstripe/silverstripe-cms/issues/2291
